src/test/java/com/exactpro/epfast/decoder/ascii/TestDecodeAsciiString.java

```diff
@@ -10,164 +12,254 @@
+
+        @Test
+        void testMandatoryEmptyString() throws IOException {
+            withByteBuf("80", buffers -> {
+                decode(decoder, buffers);
+                assertTrue(decoder.isReady());
+                assertEquals("", decoder.getValue());
+            });
+        }
+
+        @Test
+        void testZeroByteStringMandatory1() throws IOException {
+            withByteBuf("00 00 80", buffers -> {
+                decode(decoder, buffers);
+                assertTrue(decoder.isReady());
+                assertEquals("\0\0", decoder.getValue());
+            });
+        }
+
+        @Test
+        void testZeroByteStringMandatory2() throws IOException {
+            withByteBuf("00 00 00 00 80", buffers -> {
+                decode(decoder, buffers);
+                assertTrue(decoder.isReady());
+                assertEquals("\0\0\0\0", decoder.getValue());
+            });
+        }
+
+        @Test
+        void testMandatoryEmptyStringGetValueTwice() throws IOException {
+            withByteBuf("80", buffers -> {
+                decode(decoder, buffers);
+                assertTrue(decoder.isReady());
+                assertEquals("", decoder.getValue());
+                assertEquals("", decoder.getValue());
+            });
+        }
+
+        @Nested
+        class TestOverflowException {
+
+            private DecodeMandatoryAsciiString decoder = new DecodeMandatoryAsciiString(true);
+
+            @Test
+            void testMandatoryOverlongT1() throws IOException {
+                withByteBuf("00 81", buffers -> {
+                    decode(decoder, buffers);
+                    assertTrue(decoder.isReady());
+                    assertThrows(OverflowException.class, () -> decoder.getValue());
+                });
+            }
+
+            @Test
+            void testMandatoryOverlongT2() throws IOException {
+                StringBuilder stringBuilder = new StringBuilder(DecodeAsciiString.MAX_ALLOWED_LENGTH + 1);
+                for (int i = 0; i < 2 * DecodeAsciiString.MAX_ALLOWED_LENGTH; i += 2) {
+                    stringBuilder.append("41 41 ");
+                }
+                stringBuilder.append("c3");
+
+                withByteBuf(stringBuilder.toString(), buffers -> {
+                    decode(decoder, buffers);
+                    assertTrue(decoder.isReady());
+                    assertThrows(OverflowException.class, () -> decoder.getValue());
+                });
+            }
+        }
+
+        @Test
+        void testMandatoryOverlong() throws IOException {
+            withByteBuf("00 81", buffers -> {
+                decode(decoder, buffers);
+                assertTrue(decoder.isReady());
+                decoder.getValue();
+            });
+        }
+
+        @Test
+        void testMandatoryMaxAllowedLength() throws IOException {
+            StringBuilder stringBuilder = new StringBuilder(DecodeAsciiString.MAX_ALLOWED_LENGTH);
```

Comment:
You do it in several places. Consider writing a method which returns string of requested length